### PR TITLE
Data loader changes

### PR DESCRIPTION
**Pull request by samjett247**
- Wrote a more terse aggregate_data function that basically does the same thing but handles errors better.
- Changed to upload all data into single collection.
- Put data into `reviews-db-v1`
- Removed question weighting mapping
- Ready for Review @schuermannator

### CODE REVIEW

**Comment by samjett247** [comment]
Happy with this PR and gonna merge in to get running on the new dataset, since I had to delete old data to make room for this. @jlovoi @schuermannator Will keep yall updated, sorry about the no chance for review.

File: data_loader.py [hunk 1]
```diff
@@ -25,97 +31,52 @@ def update_database(force_update=False):
...
+        print('Converting the scraped collection -'+ocr_coll+ '- to pd dataframe.')
         ocr_db = conn.get_db_collection(OCR_DB_NAME, ocr_coll)
         df = pd.DataFrame(list(ocr_db.find()))
-        df.drop(['_id'],axis=1, inplace=True)
-        df.rename(columns ={'Individual Responses':'Responses'}, inplace=True)
+
+        # Condition the df prior to aggregation
+        df = df.drop(['_id'],axis=1, errors = 'ignore').rename(columns ={'Individual Responses':'Responses'})
         df['Instructor ID'] = (df['Instructor First Name']+df['Instructor Last Name']).apply(str).apply(hash).astype('int32').abs()
+        df['course_uuid'] = (df['Subject Code']+df['Course Number'].apply(str)+df['Section Title'].apply(lambda x: x[:-4])).apply(str).apply(hash).astype('int32').abs().apply(str)
```
Comment (zachschuermann) [inline data_loader.py:45]: funcshunal

File: data_loader.py [hunk 2]
```diff
@@ -1,21 +1,27 @@
+# global/PyPI
 import pandas as pd
-from mongo import mongo_driver as db_conn
 import json
 import os
 import hashlib
+from tqdm import tqdm
 
+# local
```
Comment (zachschuermann) [inline data_loader.py:8]: eh.. del and we should just follow PEP import semantics

File: mappings.yaml [hunk 3]
```diff
@@ -151,10 +41,39 @@ Term_code_dict: {'201010': 'Fall 2010',
  '202310': 'Fall 2023',
  '202320': 'Spring 2023',
  '202330': 'Summer 2023'} 
- 
 
-College_code_dict: {'EN': 'Gallogly College of Engineering', 'ED':'Jeanine Rainbolt College of Education'}
+# These map the headers in the college to the short names in the db. 
+# Note that short names are just first char of each word in the College Name, w/o spaces.
```
Comment (zachschuermann) [inline mappings.yaml:46]: nice

File: mappings.yaml [hunk 4]
```diff
@@ -151,10 +41,39 @@ Term_code_dict: {'201010': 'Fall 2010',
...
+'Jeannine Rainbolt College of Education': 'JRCoE',
+'Gallogly College of Engineering': 'GCoE',
+'Weitzenhoffer Family College of Fine Arts': 'WFCoFA',
+'Honors College': 'HC', 'College of International Studies': 'CoIS',
+'Gaylord College of Journalism and Mass Communication': 'GCoJaMC',
+'College of Professional and Continuing Studies': 'CoPaCS',
+'University College': 'UC', 'Center for Independent and Distance Learning': 'CfIaDL',
+'Expository Writing Program': 'EWP', 'ROTC - Air Force': 'R-AF'}
 
+# DEPRECATED
```
Comment (zachschuermann) [inline mappings.yaml:62]: fine to delete I think

File: tests/test.py [hunk 5]
```diff
@@ -3,22 +3,25 @@
...
+
 class basictest(unittest.TestCase):
     """ Basic tests """
-
     # Test for mongo.py
     def test_connection(self):
         '''
         This unittest will test whether the mongo driver is connecting successfully to:
-        Database name: "reviews-db"
+        Database name:
```
Comment (zachschuermann) [inline tests/test.py:20]: DB_NAME

File: tests/test.py [hunk 6]
```diff
@@ -122,11 +138,20 @@ def test_zdataframe_aggregation(self): # Z puts it last
...
+        df['Instructor First Name'] = df['Instructor First Name'].apply(str.title)
+        df['Instructor Last Name'] = df['Instructor Last Name'].apply(str.title)
 
         ag_df = data_aggregation.aggregate_data(df)
 
         # There should be no entries with the same course name, Instructor ID, and Term Code, so the below should be false
-        num_repeats = len(ag_df[ag_df[['course_uuid', 'Term Code','Instructor ID']].duplicated() == True])
+        num_repeats = len(ag_df[ag_df[['Subject Code', 'Course Number', 'Course Title', 'Term Code','Instructor ID']].duplicated() == True])
 
         return self.assertEqual(0, num_repeats)
```
Comment (zachschuermann) [inline tests/test.py:157]: nice, thanks for doing tests.

File: data_aggregation.py [hunk 7]
```diff
@@ -1,245 +1,82 @@
...
-    ag_df.insert(12, 'Avg Instructor Rating In Section', 0.0)
-    ag_df.insert(13, 'SD Instructor Rating In Section', 0.0)
-    ag_df.insert(15, 'Course Enrollment', 0)
-
-    # Rename the necessary columns
-    ag_df.rename(columns = {'Section Title':'Course Title', 'Responses':'Instructor Enrollment'}, inplace= True)
+    """
+    Aggregates a pandas dataframe of student reviews data. See the First, Second, and Third Operations below for more descriptions of functions.
+    Note the similar form across operations.
+    :Inputs:
```
Comment (zachschuermann) [inline data_aggregation.py:35]: Idc what docstrings we use really, but might be a good idea to just do PEP. I think they endorse reST. 

Here:
```python3
"""
This is a reST style.

:param param1: this is a first param
:param param2: this is a second param
:returns: this is a description of what is returned
:raises keyError: raises an exception
"""
```

**Review by zachschuermann — COMMENTED** [review]
Looks great. I know this is a post-merge review, but I just wanted to do it for posterity. Don't feel like you need to make any of the changes. We can talk about them soon; just minor stuff I figured I'd share. Thanks for the hard work :)

**Comment by zachschuermann** [comment]
Again, did a review, but not really necessary - @samjett247 delete branch whenever you're ready.

**Comment by samjett247** [comment]
Thanks Schuerm, lets talk about some other small changes we could include with GQL stuff. It's really minor but we need to change the hash function to be deterministic for mapping given input to output. See #32

File: data_aggregation.py (hunk 7 above)
Comment (samjett247) [reply data_aggregation.py:35]: Appreciate the PEP stuff; Don't have much experience following standards in developing software but its v helpful.

**Pull request merged**